profiling/src/lib.rs

```diff
@@ -1045,6 +1061,43 @@ unsafe fn restore_zend_heap(heap: *mut zend::_zend_mm_heap, custom_heap: c_int)
     std::ptr::write(heap as *mut c_int, custom_heap);
 }
 
+#[cfg(feature = "allocation_profiling")]
+unsafe extern "C" fn datadog_allocation_profiling_gc_mem_caches(
+    execute_data: *mut zend::zend_execute_data,
+    return_value: *mut zend::zval,
+) {
+    // Better safe than sorry: `gc_mem_caches()` is a Zend core function which
+    // always exists
+    if GC_MEM_CACHES_HANDLER == None {
```

Comment:
wouldn't something like `let Some(func) = GC_MEM_CACHES_HANDLER else {` be much more idiomatic than having Safety annotations below?